lib/tasks/match_identity_with_professional_organization.rake

```diff
@@ -0,0 +1,208 @@
+  puts "Number of Unmatched Identities:"
+  puts Identity.all.where(professional_organization_id: nil).count
+  ### END DEPARTMENT ###
+
+  ### COLLEGE ###
+  one_college_match = Hash.new
+  more_than_one_college_match = Hash.new
+
+  Identity.all.map(&:college).uniq.each do |college|
+    college_org = ProfessionalOrganization.where("LOWER(professional_organizations.name) LIKE LOWER('%#{college}%')").where(org_type: 'college')
+    if !college_org.empty?
+      if college_org.count == 1
+        one_college_match[college] = college_org.first.id
+      else
+        more_than_one_college_match[college] = college_org.map(&:id)
+      end
+    end
+  end
+
+  # Colleges were much easier to match then department and there were no more_than_one_college_match
+  # Update Identities and their professional_organization_ids according to college
+  one_college_match.each do |key, value|
+    unassigned_identities = Identity.all.where(professional_organization_id: nil)
+    identities_with_key = unassigned_identities.where(college: key) - @identities_that_need_discussion
+    identities_with_key.each do |identity_with_key|
+      identity_with_key.update_attribute(:professional_organization_id, value)
+    end
+  end
+  puts "END OF COLLEGE"
+  puts "Number of Unmatched Identities:"
+  puts Identity.all.where(professional_organization_id: nil).count
+
+  ### END COLLEGE ###
+
+  ### INSTITUTION ###
+
+  unassigned_identities = Identity.all.where(professional_organization_id: nil)
+  identities_with_key = unassigned_identities.where(institution: 'medical_university_of_south_carolina') - @identities_that_need_discussion
+  # The only institution that matched with Professional Organization Institutions was 'medical_university_of_south_carolina'
+  # Update Identities and their professional_organization_ids according to 'medical_university_of_south_carolina'
+  identities_with_key.each do |identity_with_key|
+    identity_with_key.update_attribute(:professional_organization_id, 3)
+  end
+  puts "END OF INSTITUTION"
+  puts "Number of Unmatched Identities:"
+  puts Identity.all.where(professional_organization_id: nil).count
+
+  ### END INSTITUTION ###
+
+  #### WENJUN MANUAL MATCHING ###
+  Identity.find(8045).update_attribute(:professional_organization_id, 4)
+  Identity.find(44254).update_attribute(:professional_organization_id, 4)
+
+  Identity.find(46501).update_attribute(:professional_organization_id, 16)
+  Identity.find(21330).update_attribute(:professional_organization_id, 18)
+
+  Identity.find(4958).update_attribute(:professional_organization_id, 23)
+  Identity.find(23505).update_attribute(:professional_organization_id, 23)
+
+  Identity.find(7787).update_attribute(:professional_organization_id, 26)
+  ### END WENJUN MANUAL MATCHING ###
+
+  ### CLEARING UNMATCHED IDENTITIES PER WENJUN ###
+  identities_with_all_nil_values = Identity.all.where(professional_organization_id: nil).where(college: nil).where(department: nil).where(institution: nil)
```

Comment:
You can condense the multiple where clauses into a single one for each query and if you identities_with_no_professional_organization_id to the top it could be slightly more efficient:
identitites_with_no_professional_organization_id = Identity.all.where(professional_organization_id: nil)
identities_with_all_nil_values = identitites_with_no_professional_organization_id.where(college: nil, department: nil, institution: nil)
identities_with_all_empty_values = identitites_with_no_professional_organization_id.where(college: "", department: "", institution: "")